jax/numpy/lax_numpy.py

```diff
@@ -2178,7 +2178,13 @@ def _can_call_numpy_array(x):
 
 # TODO(mattjj): maybe move these two functions into xla.py
 def _device_put_raw(x):
-  return array_result_handler(None, abstractify(x))(device_put(x))
+  try:
+    result = array_result_handler(None, abstractify(x))(device_put(x)) 
+  except:
+    pass 
```

Comment:
Instead of `pass` here, I think we could put the final line `raise TypeError("Unexpected input type for array: {}".format(type(x))) from None` here, with the `from None` on the end to avoid the extra exception printing.

Reply:
Great suggestion, the code is much better now. Also thank you for the `from None` explanation, it saved me some time :)